tests/jax/systems/mappo_system_test.py

```diff
@@ -1,30 +1,161 @@
 import functools
+from datetime import datetime
+from typing import Any, List
 
-import acme
+import launchpad as lp
 import optax
 import pytest
+from launchpad.launch.test_multi_threading import (
+    address_builder as test_address_builder,
+)
 
+from mava.components.jax.building.distributor import Distributor, DistributorConfig
+from mava.core_jax import SystemBuilder
 from mava.systems.jax import mappo
+from mava.systems.jax.launcher import Launcher, NodeType
 from mava.systems.jax.system import System
+from mava.utils import lp_utils
 from mava.utils.environments import debugging_utils
 from mava.utils.loggers import logger_utils
 
 #########################################################################
 # Full system integration test.
 
 
+class MockLauncher(Launcher):
+    def __init__(
+        self,
+        multi_process: bool,
+        nodes_on_gpu: List = [],
+        sp_trainer_period: int = 10,
+        sp_evaluator_period: int = 10,
+        name: str = "System",
+        terminal: str = "current_terminal",
+    ) -> None:
+        """Create mock launcer component in order to be able to \
+            overwrite the launch method which enables for testing \
+                a full distribured Mava system."""
+        super().__init__(
+            multi_process,
+            nodes_on_gpu,
+            sp_trainer_period,
+            sp_evaluator_period,
+            name,
+            terminal,
+        )
+
+    def launch(self) -> None:
```

Comment:
I'm not the biggest fan of mocking out these methods, since it means we're not actually testing some of the launchpad code. However, I suspect that you may have found the best way to do this, so feel free to disregard this if there's no better way to test this using the actual Mava launcher.

Reply:
I think that `Mock` is a misleading way of referring to these. I can change the naming to reflect what is really happening. In order to be able to test launchpad systems `launch_type` has to be set tp `"test_mt"` as is done [here[(https://github.com/instadeepai/Mava/blob/45e9178906d368baaa6c67bd1d6cf08cbe39ff77/tests/jax/systems/mappo_system_test.py#L57). In the [tf system integration tests](https://github.com/instadeepai/Mava/blob/45e9178906d368baaa6c67bd1d6cf08cbe39ff77/tests/tf/systems/mappo_system_test.py#L71) we were able to set this directly in the tests due to the way were calling the `lp.launch` method, but in the redesign this method is called inside the `launch` method of the `Launcher` component which gets added to the builder store as `program` by the `Distributor` component. So we are still testing the really mava launcher, we are just setting it to 'test mode'. There is a bit more of a discussion about it [on the launchpad repo](https://github.com/deepmind/launchpad/blob/master/docs/get_started.md).

Reply:
Another way, which would remove the need for doing this entirely, would be to add a `test` parameter to the actual `Launcher` component but I opted for not changing a component in order to suit the needs of the tests in this case.

Reply:
Happy with the existing solution now that comments have been added. As discussed: we don't think it's a good idea to change the launcher to make it more testable.